File: app/src/main/java/com/khmelenko/lab/varis/network/retrofit/raw/RawApiService.java
package com.khmelenko.lab.varis.network.retrofit.raw;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;


/**
 * @author Dmytro Khmelenko
 */
public interface RawApiService {

    @GET("/jobs/{jobId}/log")
    Single<String> getLog(@Header("Authorization") String basicAuth, @Path("jobId") String jobId);

    @GET("/jobs/{jobId}/log")
    Single<String> getLog(@Path("jobId") String jobId);
}

File: app-v3/src/main/java/com/khmelenko/lab/varis/network/retrofit/raw/RawApiService.kt
package com.khmelenko.lab.varis.network.retrofit.raw

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Path


/**
 * @author Dmytro Khmelenko
 */
interface RawApiService {

    @GET("/jobs/{jobId}/log")
    fun getLog(@Header("Authorization") basicAuth: String, @Path("jobId") jobId: String): Single<String>

    @GET("/jobs/{jobId}/log")
    fun getLog(@Path("jobId") jobId: String): Single<String>
}
